lms/static/scripts/frontend_apps/components/dashboard/SyncGradesButton.tsx

```diff
@@ -0,0 +1,120 @@
+import {
+  Button,
+  LeaveIcon,
+  SpinnerCircleIcon,
+} from '@hypothesis/frontend-shared';
+import { useCallback, useMemo, useState } from 'preact/hooks';
+import { useParams } from 'wouter-preact';
+
+import type { GradingSync } from '../../api-types';
+import { useConfig } from '../../config';
+import { apiCall, usePolledAPIFetch } from '../../utils/api';
+import type { QueryParams } from '../../utils/url';
+import { replaceURLParams } from '../../utils/url';
+
+export type SyncGradesButtonProps = {
+  /**
+   * List of students and their grades, which should be synced when the button
+   * is clicked.
+   * Passing `undefined` means students are not yet known and/or being loaded.
+   */
+  studentsToSync?: Array<{ h_userid: string; grade: number }>;
+};
+
+export default function SyncGradesButton({
+  studentsToSync,
+}: SyncGradesButtonProps) {
+  const { assignmentId } = useParams<{ assignmentId: string }>();
+  const { dashboard, api } = useConfig(['dashboard', 'api']);
+  const { routes } = dashboard;
+
+  const syncUrl = useMemo(
```

Comment:
```suggestion
  const syncURL = useMemo(
```

Reply:
I promise I will eventually remember this 😅